### PR TITLE
Add Openshift Client and List Functions

### PR DESCRIPTION
**Pull request by soggiest**
This commit adds an Openshift client to the Cilium-CLI K8s client and adds associated list commands. It seemed cleaner to create a dedicated Openshift client and add that to `KubernetesClient`. This keeps the `k8s/client.go` cleaner and easier to read.

The commit also adds mock Openshift functions. Tests can be added later.

All changes have been tested against an Openshift cluster.

The following Openshift clients are not included:
* authorization - Likely to expose secure data
* console - Not necessary for sysdump
* helm - Not necessary for sysdump
* imageregistry - Not necessary for sysdump
* insights - Not necessary for sysdump
* monitoring - Not necessary for sysdump
* oauth - Likely to expose secure data
* operatorcontrolplane - doesn't do anything
* servicecertsigner - v1alpha -  Not necessary for sysdump
* sharedResource - v1alpha1 -  Not necessary for sysdump
* template - Not necessary for sysdump
* user - Likely to expose secure data

### CODE REVIEW

File: k8s/client.go [hunk 1]
```diff
@@ -47,16 +47,32 @@ import (
 	tetragonv1alpha1 "github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
 	tetragonClientset "github.com/cilium/tetragon/pkg/k8s/client/clientset/versioned"
 
+	//openshiftAPI "github.com/openshift/api"
```
Comment (sayboras) [inline k8s/client.go:50]: this can be removed?

File: openshift-deps.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (sayboras) [inline openshift-deps.sh:1]: is this one required ?

**Review by sayboras — COMMENTED** [review]
Can you fix go mod e.g. go mod tidy && do mod vendor?

File: openshift-deps.sh (hunk 2 above)
Comment (soggiest) [reply openshift-deps.sh]: It is not, I meant to remove it before submitting. I've pushed a commit to remove it.

File: k8s/client.go (hunk 1 above)
Comment (soggiest) [reply k8s/client.go]: Yup, removed it now

**Comment by soggiest** [comment]
> Can you fix go mod e.g. go mod tidy && do mod vendor?

I've fixed go.mod

**Review by nathanjsweet — APPROVED** [review]

**Comment by tklauser** [comment]
> It would be awesome to see this merged anytime soon grin wink .

@soggiest the PR currently has some merge conflicts. Could you please rebase the PR's branch on latest `main`?

File: k8s/openshift.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
+package k8s
+
+import (
+	oseAPI "github.com/openshift/api"
+
+	//openshiftBuild "github.com/openshift/api/build"
```
Comment (tklauser) [inline k8s/openshift.go:6]: Please remove this.

**Review by sayboras — COMMENTED** [review]
Can you help to rebase ?

**Pull request closed without merging**